library/src/main/java/com/getstream/sdk/chat/model/Channel.java

```diff
@@ -392,9 +392,11 @@ public void setReactionTypes(Map<String, String> reactionTypes) {
 
     @Override
     public boolean equals(@Nullable Object obj) {
-        if (getClass() != obj.getClass()) {
+        if (obj == null) {
             return false;
         }
+        if (!(obj instanceof Channel))
```

Comment:
That is still invalid actually: hashCode must be overridden also, because Sets ( and HashSets and other collections and util classes) use hashCode to have reference to the object. There is no need to write manually equals/hashCode methods. IDE can generate it (with cmd + N). Also comparison with the current object is missed, so this one is correct and generated version:
```java
@Override
public boolean equals(Object o) {
    if (this == o) return true;
    if (o == null || getClass() != o.getClass()) return false;
    Channel channel = (Channel) o;
    return Objects.equals(cid, channel.cid);
}

@Override
public int hashCode() {
    return Objects.hash(cid);
}
```